### ISSUE TITLE
Shared ERROR and WARNING codes

### ISSUE THREAD

**Issue by edeutsch** [NCATSTranslator/ReasonerAPI#211]
It seems beneficial to have a shared collection of ERROR and WARNING codes that we may reuse amongst tools.

An initial attempt and a strawman draft of such a collection is available for comment here:
https://docs.google.com/document/d/1DPVsdjmyPzGuZsLxADio_dt4Xun98ANHb689u_n9jEU

Please read and comment

**Comment by ehinderer** [comment]
I agree on the benefit to having something like this. It ties in with the Quality Management System discussions. 

Looking through the draft, QueryGraphUnsupported looks like it might encompass QueryGraphTooManyHops. Not sure if the "unsupported" is meant to be more of a catch-all or not. By "topology" do you mean instances where there might be bifurcated query graphs?

**Comment by edeutsch** [comment]
QueryGraphTooManyHops is quite specific because some services may only permit single-hop queries
QueryGraphUnsupported is more generic that something about the query graph doesn't make sense to the agent (although it may seem perfectly valid to a human) and the query cannot go forward. Maybe bifurcated query graphs, maybe circular query_graphs may not be supported, whatever.

**Comment by vgardner-renci** [comment]
@colleenXu Can you provide more context?

**Comment by colleenXu** [comment]
I wanted clarification on whether the work on this from the May Translator Relay [here](https://drive.google.com/drive/folders/1vyHSq0xpj8aZs9OkAUBWFHt4QrrHqxwe) is complete and should be implemented by all Translator components. 

That session of the relay seemed to be a continuation of the discussion here, but it was unclear to me if the discussion was completed / decisions were finalized...

**Comment by edeutsch** [comment]
I think it's a good idea, but enthusiasm seemed muted and it is probably not so urgent. I suggest we return to this topic after 1.2-beta drops.
